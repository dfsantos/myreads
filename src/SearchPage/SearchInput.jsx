import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onChange: PropTypes.func.isRequired,
};

function SearchInput({ onChange }) {
  return (
    <div className="search-books-input-wrapper">
      <input type="text" placeholder="Search by title or author" onChange={onChange} />
    </div>
  );
}

SearchInput.propTypes = propTypes;

export default SearchInput;
