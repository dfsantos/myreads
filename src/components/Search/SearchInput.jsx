import React from 'react';
import PropTypes from 'prop-types';

import { Debounce } from 'react-throttle';

const propTypes = {
  onChange: PropTypes.func.isRequired,
};

function SearchInput({ onChange }) {
  return (
    <div className="search-books-input-wrapper">
      <Debounce time="500" handler="onChange">
        <input type="text" placeholder="Search by title or author" onChange={onChange} />
      </Debounce>
    </div>
  );
}

SearchInput.propTypes = propTypes;

export default SearchInput;
