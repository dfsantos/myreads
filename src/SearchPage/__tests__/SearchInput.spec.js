import React from 'react';
import toJson from 'enzyme-to-json';

import SearchInput from '../SearchInput';

const onChangeListener = jest.fn();

it('should render without errors', () => {
  shallow(<SearchInput onChange={onChangeListener} />);
});

it('should call onChangeListener when value change', () => {
  const wrapper = shallow(<SearchInput onChange={onChangeListener} />);
  const input = wrapper.find('input');
  const event = { target: { value: 'android' } };

  input.simulate('change', event);

  expect(onChangeListener).toHaveBeenCalled();
});

it('default snpashot', () => {
  const wrapper = shallow(<SearchInput onChange={onChangeListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
