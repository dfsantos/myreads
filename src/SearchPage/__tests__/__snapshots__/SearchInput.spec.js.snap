// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default snpashot 1`] = `
<div
  className="search-books-input-wrapper"
>
  <Debounce
    handler="onChange"
    handlers={Array []}
    time="500"
  >
    <input
      onChange={[Function]}
      placeholder="Search by title or author"
      type="text"
    />
  </Debounce>
</div>
`;
